ezGroceriesShoppingList/src/test/java/com/ezgroceries/shoppinglist/WebControllerTests.java

```diff
@@ -0,0 +1,129 @@
+package com.ezgroceries.shoppinglist;
+
+import com.ezgroceries.shoppinglist.entity.Cocktail;
+import com.ezgroceries.shoppinglist.entity.ShoppingList;
+import com.ezgroceries.shoppinglist.service.CocktailServiceImpl;
+import com.ezgroceries.shoppinglist.service.ShoppingListService;
+import com.ezgroceries.shoppinglist.web.WebController;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.List;
+
+import static com.ezgroceries.shoppinglist.util.TestData.*;
+import static com.ezgroceries.shoppinglist.util.TestHelperMethods.resourceToString;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(WebController.class)
+@AutoConfigureMockMvc(addFilters = false)
+public class WebControllerTests {
+
+
+    @MockBean
+    CocktailServiceImpl cocktailService;
+
+    @MockBean
+    ShoppingListService shoppingListService;
+
+    @SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
```

Comment:
Interesting

Reply:
It gave me an annoying warning that no beans are found. I researched it and learned that this is a problem for certain versions of Intellij and this is the solution.